### ISSUE TITLE
doc: write document that compares Upspin to other systems

### ISSUE THREAD

**Issue by adg** [upspin/upspin#177]
There are a few systems similar to Upspin out there. (Camlistore, Tahoe-LAFS, KeybaseFS, etc.) Each of them make their own set of design trade-offs. We should have a page that explains how Upspin differs to them.

**Comment by niutech** [comment]
Exactly. Why don't you join some established project (the closest to your vision) rather than creating n+1 standard? You want to be in control, right? Isn't it NIH?

**Comment by adg** [comment]
@niutech I don't really appreciate the uncharitable insinuations. We're not megalomaniacs or fools (at least, I don't think so), just a group of people with a specific vision. To my knowledge there are no other projects that make the same set of design decisions as Upspin. The fact that there are so many similar systems surely demonstrates that there are many ways to slice this pie. Thanks.

**Comment by niutech** [comment]
Thank you for you reply. I did not meant to offend you. I just wanted
to know the rationale behind your project when there are lots of
existing solutions.
I observe this phenomenon with programming frameworks, CMS-es and even
Linux distributions - almost every company builds its own, increasing
fragmentation, rather than join forces.

**Comment by robpike** [comment]
From my perspective, these terms are the ones that matter. No other system combines them this way.

_Universal_: Everyone is welcome, everyone can see the whole space (given permission). It's just one big space.
_Secure_: Full end-to-end encryption. You can store your data in public, such as with a cloud provider, with confidence.
_Sharable_: Fine-grained securing sharing with only the people you want to share with. You can share with your family, or your friends, or whatever specific set of folks you want. You are in charge of who sees what.

**Comment by wernight** [comment]
On https://github.com/upspin/upspin/README or on https://upspin.io/doc/overview.md it could quickly explain:

 - Where is the data stored? (for example reading the example of Facebook to Twitter, what if you host your data on your machine, then you Facebook would need to proxy and self-host a copy of your data; so please briefly address this issue there)
 - How is that different from Drive/DropBox/Photos/...? All these cloud solution have CLI interfaces and a URL can be seen as a unique name to a user's resource (case of Google Photos for example).

For example Google Drive can also claim that everyone is welcome, that it's secure, and shareable.

**Comment by edpin** [comment]
Here are some links that may or may not be directly related, but are worth a look as they appear to be relevant in one aspect or another:

https://mountainduck.io/
https://owncloud.org/
https://nextcloud.com/
https://nofile.io/
https://syncthing.net/
http://rclone.org/

**Comment by shoeper** [comment]
https://www.seafile.com/en/home/ would also fit into that list.

**Comment by fazalmajid** [comment]
Also https://bazil.org/ but the last update was in 2015. Upspin uses some of their code (e.g. FUSE), not sure if there are shared people or not.
